catalog/plugin/catalog-plugin-query-monitor-commands/src/main/java/ddf.catalog.querymonitorcommands/QueryMonitorCommandPrintTable.java

```diff
@@ -0,0 +1,187 @@
+
+    private static final int TIME_MAX_LENGTH = 30;
+
+    private static final int UUID_MAX_LENGTH = 37;
+
+    private String formatString =
+            "%1$-" + USER_MAX_LENGTH + "s %2$-" + SOURCE_MAX_LENGTH + "s %3$-" + SEARCH_MAX_LENGTH
+                    + "s %4$-" + TIME_MAX_LENGTH + "s %5$-" + UUID_MAX_LENGTH + "s %n";
+
+    PrintStream console = null;
+
+    PrintStream getPrintStream() {
+        return System.out;
+    }
+
+    public QueryMonitorCommandPrintTable(QueryMonitorPlugin q) {
+        queryMonitor1 = q;
+        console = getPrintStream();
+    }
+
+    public QueryMonitorCommandPrintTable() {
+        console = getPrintStream();
+    }
+
+    /**
+     * Method to implement {@link Action}. Called when the catalog:printtable command is called by the shell.
+     * Prints out a table representing a {@link QueryMonitorPlugin}'s ActiveSearchTable including information
+     * on all {@link ActiveSearch}'s currently populating the table
+     *
+     * @return always returns {@code null}
+     */
+
+    @Override
+    public Object execute() throws Exception {
+        try {
+            printSearchTableToConsole();
+        } catch (Exception e) {
+            LOGGER.debug(
+                    "Exception encountered in doExecute of QueryMonitorCommandPrintActiveSearcHTable.java!"
+                            + e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Helper method called by the execute method when the catalog:printtable command is called by the shell.
+     * Prints out a table representing a {@link QueryMonitorPlugin}'s ActiveSearchTable including information
+     * on all {@link ActiveSearch}'s currently populating the table
+     */
+
+    void printColor(Ansi.Color color, String message) {
+        String colorString;
+        if (color == null || color.equals(Ansi.Color.DEFAULT)) {
+            colorString = Ansi.ansi()
+                    .reset()
+                    .toString();
+        } else {
+            colorString = Ansi.ansi()
+                    .fg(color)
+                    .toString();
+        }
+        console.print(colorString);
+        console.print(message);
+        console.println(Ansi.ansi()
+                .reset()
+                .toString());
+    }
+
+    /**
+     * Called by the execute method when the catalog:printtable command is called by the shell.
+     * Prints out a table representing a {@link QueryMonitorPlugin}'s ActiveSearchTable including information
+     * on all {@link ActiveSearch}'s currently populating the table
+     */
+
+    void printSearchTableToConsole() {
+
+        if (queryMonitor1 == null) {
+            LOGGER.error(
+                    "Cannot print search table. QueryMonitorImpl not yet instantiated! Cannot printSearchTableToConsole()!");
+            return;
+        }
+        HashMap<UUID, ActiveSearch> activeSearchTable = queryMonitor1.getSearchTable();
+        if (activeSearchTable == null) {
```

Comment:
It doesn't look like your query monitor plugin implementation of `getSearchTable()` can return null. If it's somehow possible for something to go wrong while getting the active search table, it may be better to throw an exception than return null.